specifyweb/frontend/js_src/lib/components/Header/Notifications.tsx

```diff
@@ -42,10 +38,21 @@ export function Notifications({
     if (notificationCount === 0) handleClose();
   }, [notificationCount, handleClose]);
 
-  React.useEffect(() => {
-    let pullInterval = INITIAL_INTERVAL;
-    let lastFetchedTimestamp: Date | undefined;
+  let pullInterval = INITIAL_INTERVAL;
+  let lastFetchedTimestamp: Date | undefined;
```

Comment:
this is not going to work
remember how react component work?

this function is called on each render! your variable is re-declared on each render
if you need your value to persist between renders you have to either:
1. use useState
2. use useRef
3. OR put the value in a closure (declare your `let` variable inside of useEffect like it was before!)

option 3 is the way to go here

Reply:
all of these variables should be moved into the hook